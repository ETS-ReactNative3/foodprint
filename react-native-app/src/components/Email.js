import React from 'react';
import { Input } from 'react-native-elements';

const Email = ({ setEmail, email, nextFieldRef }) => {
  return (
    <Input
      // label={'Your Email Address'}
      autoCorrect={false}
      autoCapitalize={'none'}
      placeholder="Email"
      value={email} returnKeyType={'next'}
      onChangeText={value => setEmail(value)}
      onSubmitEditing={() => nextFieldRef.current.focus()}
      containerStyle={{ marginBottom: 15 }}
    />
  );
};

export default Email;
