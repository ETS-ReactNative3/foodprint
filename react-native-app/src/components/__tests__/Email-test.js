import React from 'react';
import Email from '../Email';

import renderer from 'react-test-renderer';

test('Email renders correctly and is the same as its previous snapshot', () => {
  const tree = renderer.create(<Email/>).toJSON();
  expect(tree).toMatchSnapshot();
});

