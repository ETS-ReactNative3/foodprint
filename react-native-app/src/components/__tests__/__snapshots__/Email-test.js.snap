// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email matches previous snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCompleteType="email"
      autoCorrect={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "alignSelf": "center",
          "color": "black",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="RNE__Input__text-input"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
